File: 2025-12/src/day_11/boj_2417/Main.java
package day_11.boj_2417;

import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        long n = Long.parseLong(br.readLine());

        System.out.println(lowerBound(n));
    }

    static long lowerBound(long key) {
        long left = 0;
        long right = (long) Math.sqrt(Long.MAX_VALUE) + 1;

        while (left < right) {
            long mid = (left + right) / 2;

            if (mid * mid < key) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return right;
    }
}

File: 2025-12/src/day_11/boj_2417/Main.cpp
#include <bits/stdc++.h>
using namespace std;

long long lower_bound_param(long long key) {
    long long left = 0;
    long long right = sqrt(LLONG_MAX) + 1;

    while (left < right) {
        long long mid = (left + right) / 2;

        if (mid * mid < key) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }

    return right;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    long long n;
    cin >> n;
    cout << lower_bound_param(n);
}
